query.go

```diff
@@ -53,7 +53,8 @@ func query(url, token, dns_name string, duration time.Duration) string {
 			resp, err = client.Do(req)
 
 			if err != nil {
-				clog.Fatalf("HTTP Error %v", err)
+				clog.Errorf("HTTP Error: %v", err)
+				return ""
```

Comment:
Returning here on Error means that you won't try again in a case of network connectivity problem.

Reply:
Heh, till now you've just called exit and stopped the daemon in case of this error! ;)
IMHO, in case of network problem it would be better to return nothing rapidly so a caller could try another way of resolving the name and NOT block the caller for 10 seconds waiting that (maybe!) the connectivity issue will be solved. If (normally) there are multiple instances of CoreDNS started and used by clients - maybe another CoreDNS servers still have connection to Netbox, so fast return from non working server will be appreciated by clients.